openstudiocore/src/model/SiteGroundTemperatureDeep.cpp

```diff
@@ -0,0 +1,653 @@
+    bool result = setDouble(OS_Site_GroundTemperature_DeepFields::NovemberDeepGroundTemperature, novemberDeepGroundTemperature);
+    OS_ASSERT(result);
+  }
+
+  void SiteGroundTemperatureDeep_Impl::resetNovemberDeepGroundTemperature() {
+    bool result = setString(OS_Site_GroundTemperature_DeepFields::NovemberDeepGroundTemperature, "");
+    OS_ASSERT(result);
+  }
+
+  void SiteGroundTemperatureDeep_Impl::setDecemberDeepGroundTemperature(double decemberDeepGroundTemperature) {
+    bool result = setDouble(OS_Site_GroundTemperature_DeepFields::DecemberDeepGroundTemperature, decemberDeepGroundTemperature);
+    OS_ASSERT(result);
+  }
+
+  void SiteGroundTemperatureDeep_Impl::resetDecemberDeepGroundTemperature() {
+    bool result = setString(OS_Site_GroundTemperature_DeepFields::DecemberDeepGroundTemperature, "");
+    OS_ASSERT(result);
+  }
+
+  void SiteGroundTemperatureDeep_Impl::setTemperatureByMonth(int month, double temperature) {
+    switch(month) {
+      case 1 : setJanuaryDeepGroundTemperature(temperature); break;
+      case 2 : setFebruaryDeepGroundTemperature(temperature); break;
+      case 3 : setMarchDeepGroundTemperature(temperature); break;
+      case 4 : setAprilDeepGroundTemperature(temperature); break;
+      case 5 : setMayDeepGroundTemperature(temperature); break;
+      case 6 : setJuneDeepGroundTemperature(temperature); break;
+      case 7 : setJulyDeepGroundTemperature(temperature); break;
+      case 8 : setAugustDeepGroundTemperature(temperature); break;
+      case 9 : setSeptemberDeepGroundTemperature(temperature); break;
+      case 10 : setOctoberDeepGroundTemperature(temperature); break;
+      case 11 : setNovemberDeepGroundTemperature(temperature); break;
+      case 12 : setDecemberDeepGroundTemperature(temperature); break;
+    }
+  }
+
+  void SiteGroundTemperatureDeep_Impl::resetTemperatureByMonth(int month) {
+    switch(month) {
+      case 1 : resetJanuaryDeepGroundTemperature(); break;
+      case 2 : resetFebruaryDeepGroundTemperature(); break;
+      case 3 : resetMarchDeepGroundTemperature(); break;
+      case 4 : resetAprilDeepGroundTemperature(); break;
+      case 5 : resetMayDeepGroundTemperature(); break;
+      case 6 : resetJuneDeepGroundTemperature(); break;
+      case 7 : resetJulyDeepGroundTemperature(); break;
+      case 8 : resetAugustDeepGroundTemperature(); break;
+      case 9 : resetSeptemberDeepGroundTemperature(); break;
+      case 10 : resetOctoberDeepGroundTemperature(); break;
+      case 11 : resetNovemberDeepGroundTemperature(); break;
+      case 12 : resetDecemberDeepGroundTemperature(); break;
+    }
+  }
+
+  void SiteGroundTemperatureDeep_Impl::resetAllMonths() {
+    for (int i = 1; i <= 12; ++i) {
+      resetTemperatureByMonth(i);
+    }
+  }
+
+  void SiteGroundTemperatureDeep_Impl::setAllMonthlyTemperatures(const std::vector<double> &monthly_temperatures) {
+    OS_ASSERT(monthly_temperatures.size() == 12);
```

Comment:
Yeah false means something went wrong and true is things worked out.  A lot of other APIs return 0 for success and an error code otherwise, we have been using exceptions instead of error codes